oaknational/Oak-Web-Application
# feat(optionality): add optionality components to units

## Description

Music year: {owa_music_year}

- List of changes

handle units with optionality
uses the optionality item compoents

### Implementation details

Plan:

- The query will return null univariants alongside their optionality counterparts
- In the query after parsing the schema:
    - Group null unitvariants with optionality variants
    - Where optionality accompanies a null variant:
        - filter out null unitvariants
        - use optionality listing item to present multiple options together
        - NB. it is possible to have a single optionality uv too
        - NB. we need to watch out for unitvariants that have programme factors but aren’t null (the MV might already do this)

### Acceptance criteria

- [x]  optionality unitvariants are correctly rendered using the oak optionality listing component
- [x]  all other unitvariants are uneffected

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2443--oak-web-application.netlify.thenational.academy/pupils/beta/programmes/english-secondary-year-10-edexcel/units
2. Click on optionality units
3. You should see all optionality units with correct links

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
